cf/data/data.py

```diff
@@ -7542,98 +7542,69 @@ def argmax(self, axis=None, unravel=False):
-                a subset of the axes.
+
+                If True then when locating the maximum over the whole
+                data, return the location as a index for each axis as
+                a `tuple`. By default an index to the flattened array
+                is returned in this case. Ignored if locating the
+                maxima over a subset of the axes.
 
         :Returns:
 
-            `int` or `tuple` or `Data`
+            `Data` or `tuple`
                 The location of the maximum, or maxima.
 
-        **Examples:**
-
-        >>> d = cf.Data(numpy.arange(120).reshape(4, 5, 6))
-        >>> d.argmax()
-        119
-        >>> d.argmax(unravel=True)
-        (3, 4, 5)
-        >>> d.argmax(axis=0)
-        <CF Data(5, 6): [[3, ..., 3]]>
-        >>> d.argmax(axis=1)
-        <CF Data(4, 6): [[4, ..., 4]]>
-        >>> d.argmax(axis=2)
-        <CF Data(4, 5): [[5, ..., 5]]>
-
-        """
-        if axis is not None:
-            ndim = self._ndim
-            if -ndim - 1 <= axis < 0:
-                axis += ndim + 1
-            elif not 0 <= axis <= ndim:
-                raise ValueError(
-                    "Can't argmax: Invalid axis specification: Expected "
-                    "-{0}<=axis<{0}, got axis={1}".format(ndim, axis)
-                )
-
-            if ndim == 1 and axis == 0:
-                axis = None
-        # --- End: if
-
-        if axis is None:
-            config = self.partition_configuration(readonly=True)
-
-            out = []
-
-            for partition in self.partitions.matrix.flat:
-                partition.open(config)
-                array = partition.array
-                index = np.unravel_index(array.argmax(), array.shape)
-                mx = array[index]
-                index = [x[0] + i for x, i in zip(partition.location, index)]
-                out.append((mx, index))
-                partition.close()
-
-            mx, index = sorted(out)[-1]
+        **Examples**
 
-            if unravel:
-                return tuple(index)
+        >>> d = cf.Data(np.arange(6).reshape(2, 3))
+        >>> print(d.array)
+        [[0 1 2]
+         [3 4 5]]
+        >>> a = d.argmax()
+        >>> a
+        <CF Data(): 5>
+        >>> a.array
+        5
+
+        >>> index = d.argmax(unravel=True)
+        >>> index
+        (1, 2)
+        >>> d[index]
+        <CF Data(1, 1): [[5]]>
 
-            return np.ravel_multi_index(index, self.shape)
+        >>> d.argmax(axis=0)
+        <CF Data(3): [1, 1, 1]>
+        d.argmax(axis=1)
+        <CF Data(2): [2, 2]>
 
-        # Parse axis
-        ndim = self._ndim
-        if -ndim - 1 <= axis < 0:
-            axis += ndim + 1
-        elif not 0 <= axis <= ndim:
-            raise ValueError(
-                "Can't argmax: Invalid axis specification: Expected "
-                "-{0}<=axis<{0}, got axis={1}".format(ndim, axis)
-            )
+        Only the first occurrence is returned.
```

Comment:
I think this note should also be covered in the description somehow, rather than only (somewhat hidden) here in the examples. Maybe adding a few words to the second sentence RE use without provision of an axis: "... locates the _first occurrence of the_ maximum of the whole data."

Also, if this stays, perhaps use a colon, or otherwise, to emphasise you are introducing/describing the example afterwards (I for one was a bit confused as to why the sentence was there on its own)?

```suggestion
        Only the first occurrence is returned:
```

Reply:
Thanks. How about 927aa0b54 ?

Reply:
Perfect, thanks. Marking as resolved.